README.md

```diff
@@ -1,131 +1,171 @@
+(ECC: Everyone Can Create, CC: Can Create Allowlist, ECG: Everyone Can Grant, CG: Can Grant Allowlist)
+- Only badge admins can create and grant badges: ECC false, ECG false, CC empty and CG empty.
+- Only user1 can create badges, but everyone can grant them (for example for peer appreciation badges, like "Thank you" badge): ECC false, ECG true, CC user1, CG empty
+- Only user1 can create badges, and only user2 and user3 can grant them (lead appreciation badges, like "MVP" badge, where the management create the badges, and the team leads are the ones granting them to their team members): ECC false, ECG false, CC user1, CG user2, user3
+- Only user1 and user2 can create badges, but they can only grant the badges they have created (this can be used to have team specific badges without creating a new type for every team): ECC false, ECG false, CC user1, user2, CG empty
+- Everyone can create badges, but can only grant the badges they have created: ECC true, ECG false, CC empty, CG empty
+- Everyone can create and grant any badge: ECC true, ECG true, CC empty, CG empty
 
-This will produce a single plugin file (with support for multiple architectures) for upload to your Mattermost server:
+### Creating a badge
+In order to create a badge, you must decide the type of that badge, and have permissions to create a badge of that type.
+Run the slash command `/badges create badge` to open the creation dialog.
 
-```
-dist/com.example.my-plugin.tar.gz
-```
+![Screenshot from 2022-03-16 11-37-32](https://user-images.githubusercontent.com/1933730/158571687-4983f7e4-1cf9-4fa1-a3f1-6f80918e28e3.png)
 
-## Development
+- Name: Name of the badge.
+- Description: Description of the badge.
+- Image: Only emojis are allowed. You must input the emoji name as you would do to add it to a message (e.g. `:+1:` or `:smile:`). Custom emojis are also allowed.
+- Type: The type of the badge. This list will show only types you have permissions to create.
+- Multiple: Whether this badge can be granted more than once to the same person.
 
-To avoid having to manually install your plugin, build and deploy your plugin using one of the following options.
+### Details about Multiple
+All badges can be assigned to any number of persons. What the Multiple setting controls is whether this badge can be granted more than once to the same person. For example, a "Thank you" badge should be granteable many times (many people can be thankful to you in more than one ocassion). This badge should have the Multiple checkbox marked. But a "First year in the company" badge should be granted just once (it makes no sense to have 2 first year in the company). That badge should have the Multiple checkbox unmarked.
```

Comment:
```suggestion
All badges can be assigned to any number of people. What the **Multiple** setting controls is whether this badge can be granted more than once to the same person. For example, a "Thank you" badge should be grantable many times (many people can be thankful to you on more than one occasion), and therefore, a Thank You badge should have the **Multiple** option selected. However, a "First year in the company" badge should be granted only once since a user won't celebrate this milestone multiple times at the same company. This type of badge should have the **Multiple** option unselected.
```